tests/test_unittest/test_gtc/test_passes/test_oir_optimizations/test_caches.py

```diff
@@ -382,13 +383,18 @@ def test_fill_to_local_k_caches_section_splitting_forward():
                 sections=[
                     VerticalLoopSectionFactory(
                         interval=IntervalFactory(end=AxisBound(level=LevelMarker.END, offset=-1)),
-                        horizontal_executions__0__body=[
-                            AssignStmtFactory(
-                                left__name="foo", right__name="foo", right__offset__k=0
-                            ),
-                            AssignStmtFactory(
-                                left__name="foo", right__name="foo", right__offset__k=1
-                            ),
+                        horizontal_executions=[
+                            HorizontalExecutionFactory(
+                                body=[
+                                    AssignStmtFactory(
+                                        left__name="foo", right__name="foo", right__offset__k=0
+                                    ),
+                                    AssignStmtFactory(
+                                        left__name="foo", right__name="foo", right__offset__k=1
```

Comment:
I am wondering if this is a good test. It has a race condition, so if we would run a checker on this piece it would trigger. On the other hand, if we force ourselves to always write "correct" snippets for unittest it could result in more complicated test setups and distraction from the actual codepath that is tested. Opinions?

Reply:
I can’t see a race condition here, it’s a forward loop. Sure it does not make any computational sense to just overwrite foo here, but it is just about the read and write offsets. So this is probably the simplest code to get a write to `foo` and a read from `foo` and `foo[0, 0, 1]`.

Reply:
Sorry, I "saw" a horizontal offset.